tests/test_elasticache/test_elasticache.py

```diff
@@ -689,3 +689,283 @@ def test_list_tags_cache_cluster():
+    vpc2 = ec2_client.create_vpc(CidrBlock="10.1.0.0/16").get("Vpc")
+    subnet1 = ec2_client.create_subnet(VpcId=vpc["VpcId"], CidrBlock="10.0.0.0/24").get(
+        "Subnet"
+    )
+    subnet2 = ec2_client.create_subnet(
+        VpcId=vpc2["VpcId"], CidrBlock="10.1.0.0/24"
+    ).get("Subnet")
+
+    with pytest.raises(ClientError) as exc:
+        client.create_cache_subnet_group(
+            CacheSubnetGroupName="test-subnet-group",
+            CacheSubnetGroupDescription="Test subnet group",
+            SubnetIds=[subnet1["SubnetId"], subnet2["SubnetId"]],
+            Tags=[
+                {"Key": "foo", "Value": "bar"},
+                {"Key": "foo1", "Value": "bar1"},
+            ],
+        )
+    err = exc.value.response["Error"]
+    assert err["Code"] == "InvalidSubnet"
+
+
+@mock_aws
+def test_cache_subnet_group_with_ipv4_and_ipv6_subnets():
+    client = boto3.client("elasticache", region_name="us-east-2")
+    ec2_client = boto3.client("ec2", region_name="us-east-2")
+
+    vpc = ec2_client.create_vpc(
+        CidrBlock="10.0.0.0/16", Ipv6CidrBlock="2600:1f16:1cb1:6b00::/56"
+    ).get("Vpc")
+
+    vpc_id = vpc.get("VpcId")
+
+    subnet_dual = ec2_client.create_subnet(VpcId=vpc_id, CidrBlock="10.0.0.0/24").get(
+        "Subnet"
+    )
+    subnet_dual_id = subnet_dual.get("SubnetId")
+
+    ec2_client.associate_subnet_cidr_block(
+        SubnetId=subnet_dual_id, Ipv6CidrBlock="2600:1f16:1cb1:6b00::/60"
+    )
+
+    subnet_ipv4 = ec2_client.create_subnet(VpcId=vpc_id, CidrBlock="10.0.1.0/24").get(
+        "Subnet"
+    )
+    subnet_ipv4_id = subnet_ipv4.get("SubnetId")
+
+    resp = client.create_cache_subnet_group(
+        CacheSubnetGroupName="test-subnet-group",
+        CacheSubnetGroupDescription="Test subnet group",
+        SubnetIds=[subnet_dual_id, subnet_ipv4_id],
+        Tags=[
+            {"Key": "foo", "Value": "bar"},
+            {"Key": "foo1", "Value": "bar1"},
+        ],
+    )
+
+    if resp["CacheSubnetGroup"]["Subnets"][0]["SubnetIdentifier"] == subnet_dual_id:
+        assert resp["CacheSubnetGroup"]["Subnets"][0]["SupportedNetworkTypes"] == [
+            "ipv4",
+            "dual_stack",
+        ]
+        assert resp["CacheSubnetGroup"]["Subnets"][1]["SupportedNetworkTypes"] == [
+            "ipv4"
+        ]
+    else:
+        assert resp["CacheSubnetGroup"]["Subnets"][0]["SupportedNetworkTypes"] == [
+            "ipv4"
+        ]
+        assert resp["CacheSubnetGroup"]["Subnets"][1]["SupportedNetworkTypes"] == [
+            "ipv4",
+            "dual_stack",
+        ]
+    assert len(resp["CacheSubnetGroup"]["SupportedNetworkTypes"]) == 2
+    assert "ipv4" in resp["CacheSubnetGroup"]["SupportedNetworkTypes"]
+    assert "dual_stack" in resp["CacheSubnetGroup"]["SupportedNetworkTypes"]
+
+
+# The following test is commented out until the create_subnet moto feature
+# adds support for Ipv6Native parameter.
+
+# @mock_aws
```

Comment:
It's good to already have a test in place! That will make it easier to implement this feature when we're ready.

Instead of commenting it out, I would suggest skipping it instead. That makes it slightly easier to read:
```python
from unittest import SkipTest
def test_cache_subnet_group_with_ipv6_native_subnets():
    raise SkipTest("create_subnet() does not support Ipv6Native-parameter yet")
    .... rest of the test
```

Reply:
That's a good idea! Skipped test